### PR TITLE
Lb haproxy study

### PR DESCRIPTION
**Pull request by Jaakko-Os**
Co-authored-by: mika.koskimaki@est.tech
Co-authored-by: anwar.hassen@est.tech

### CODE REVIEW

**Comment by Jaakko-Os** [comment]
@mikkosest

File: lb-haproxy-study/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Configure HAProxy on control plane as a daemonset on top of Kubernetes or as a service on plain Linux
+
+## HAPROXY as daemonset or service
+* Assumption for all options is that needed **cloud-init** can be created from userdata given in Kubernetes CRs
+* Goal is to describe control plane load balancing options without cluster external load balancer
```
Comment (kashifest) [inline lb-haproxy-study/README.md]: cluster external?

File: lb-haproxy-study/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  * HAProxy update logic periodically reads the status of configmap and if configmap is changed, it will update master node IP addresses **/etc/haproxy/haproxy.cfg** and then restarts HAProxy gracefully by using HAProxy internal config update mechanism.
+  * Optionally changes in configmap can be mounted to host file system by using priviledged containers in Kubernetes.
+* A delay (~60 sec) is identified when configmap is modified until the HAProxy sees the change, not seen as an issue at least for now
+
+Pros: No need to port tunneling before control plane is ready as in cluster internal LB options. Cluster reboot works without any additional logic like in daemonset options.
+Cons: Loadbalancer upgrades means upgrades in Linux host services.
+
+#### HAProxy as a daemonset
+* Iptables or socat tunneling for port forwarding needed from LB port to API server port, this needs to be activated in **cloud-init** before kubeadm init
```
Comment (kashifest) [inline lb-haproxy-study/README.md]: IPtables

File: lb-haproxy-study/README.md (hunk 1 above)
Comment (Jaakko-Os) [reply lb-haproxy-study/README.md]: meaning external LB, I see this clear enough.

**Pull request merged**